### PR TITLE
feat: user history

### PR DESCRIPTION
**Pull request by kobryl**

### CODE REVIEW

File: objects_counter/api/utils.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            try:
+                data = jwt.decode(token, current_app.config['SECRET_KEY'], algorithms=['HS256'])
+                current_user = get_user_by_id(data['user_id'])
+                if current_user is None:
+                    return Response('Invalid token', HTTPStatus.UNAUTHORIZED)
+            except Exception as e:  # pylint: disable=broad-except
+                return Response({
+                    'message': 'Issue while processing authentication',
+                    'error': str(e)
+                }, HTTPStatus.INTERNAL_SERVER_ERROR)
```
[the remark this answers is not in the document]
Comment (kobryl) [reply objects_counter/api/utils.py:52]: Will be fixed in #24 - it is meant for development environment for now

File: objects_counter/api/utils.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            current_user = get_user_by_id(data['user_id'])
+            if current_user is None:
+                return Response('Invalid token', HTTPStatus.UNAUTHORIZED)
+        except jwt.ExpiredSignatureError:
+            return Response('Token expired', HTTPStatus.UNAUTHORIZED)
+        except Exception as e:  # pylint: disable=broad-except
+            return Response({
+                'message': 'Issue while processing authentication',
+                'error': str(e)
+            }, HTTPStatus.INTERNAL_SERVER_ERROR)
```
[the remark this answers is not in the document]
Comment (kobryl) [reply objects_counter/api/utils.py:29]: Will be fixed in #24 - it is meant for development environment for now

File: objects_counter/api/results/views.py [hunk 3]
```diff
@@ -1,29 +1,37 @@
...
+            return Response('Invalid result ID', 400)
+        try:
+            result = get_result_by_id(result_id)
+            if result.user_id != current_user.id:
+                return Response('Unauthorized', 403)
+            return Response(result, 200)
+        except NotFound as e:
+            return Response(str(e), 404)
+        except Exception as e:
+            return Response(str(e), 500)
```
[the remark this answers is not in the document]
Comment (kobryl) [reply objects_counter/api/results/views.py:37]: Will be fixed in #24 - it is meant for development environment for now

File: objects_counter/api/users/views.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+
+@api.route('/login', doc=False)
+class Login(Resource):
+    def post(self) -> typing.Any:
+        data = request.json
+        try:
+            username, password = get_user_from_input(data)
+        except ValueError as e:
+            return Response(str(e), status=400)
```
[the remark this answers is not in the document]
Comment (kobryl) [reply objects_counter/api/users/views.py:22]: Will be fixed in #24 - it is meant for development environment for now

File: objects_counter/api/users/views.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        if user is None:
+            return Response('Invalid username or password', status=404)
+        try:
+            token = jwt.encode({"user_id": user.id}, current_app.config['SECRET_KEY'], algorithm='HS256')
+            return Response(json.dumps({
+                'token': token,
+                'user_id': user.id
+            }), status=200, content_type='application/json')
+        except Exception as e:
+            return Response(str(e), status=500)
```
[the remark this answers is not in the document]
Comment (kobryl) [reply objects_counter/api/users/views.py:33]: Will be fixed in #24 - it is meant for development environment for now

File: objects_counter/api/users/views.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+
+@api.route('/register', doc=False)
+class Register(Resource):
+    def post(self) -> typing.Any:
+        data = request.json
+        try:
+            username, password = get_user_from_input(data)
+        except ValueError as e:
+            return Response(str(e), status=400)
```
[the remark this answers is not in the document]
Comment (kobryl) [reply objects_counter/api/users/views.py:43]: Will be fixed in #24 - it is meant for development environment for now

File: objects_counter/api/users/views.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        try:
+            user = insert_user(username, password)
+        except ValueError as e:
+            return Response(str(e), status=400)
```
[the remark this answers is not in the document]
Comment (kobryl) [reply objects_counter/api/users/views.py:47]: Will be fixed in #24 - it is meant for development environment for now

File: objects_counter/api/results/views.py [hunk 8]
```diff
@@ -1,29 +1,37 @@
...
+    def get(self, current_user: User, result_id: int) -> typing.Any:
+        if result_id < 0:
+            return Response('Invalid result ID', 400)
+        try:
+            result = get_result_by_id(result_id)
+            if result.user_id != current_user.id:
+                return Response('Unauthorized', 403)
+            return Response(result, 200)
+        except NotFound as e:
+            return Response(str(e), 404)
```
[the remark this answers is not in the document]
Comment (kobryl) [reply objects_counter/api/results/views.py:35]: Will be fixed in #24 - it is meant for development environment for now

**Review by michalkr52 — APPROVED** [review]

**Pull request merged**